bugsnag-android-core/src/test/java/com/bugsnag/android/EventDeliveryStrategyTest.kt

```diff
@@ -0,0 +1,124 @@
+import com.bugsnag.android.internal.dag.ValueProvider
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.junit.MockitoJUnitRunner
+import java.util.Date
+
+@RunWith(MockitoJUnitRunner::class)
+internal class EventDeliveryStrategyTest {
+
+    @Mock
+    private lateinit var eventStore: EventStore
+
+    private val apiKey = "BUGSNAG_API_KEY"
+    private val notifier = Notifier()
+    private val config = generateImmutableConfig()
+    private val callbackState = CallbackState()
+    private val logger = InterceptingLogger()
+    private lateinit var deliveryDelegate: DeliveryDelegate
+    private val handledState = SeverityReason.newInstance(
+        SeverityReason.REASON_UNHANDLED_EXCEPTION
+    )
+    private var testException: RuntimeException? = null
+    private val event = Event(
+        RuntimeException("Whoops!"), config, handledState, NoopLogger
+    )
+
+    @Before
+    fun setUp() {
+        testException = RuntimeException("Example message")
+        deliveryDelegate =
+            DeliveryDelegate(
+                logger,
+                ValueProvider(eventStore),
+                config,
+                callbackState,
+                notifier,
+                BackgroundTaskService()
+            )
+        event.session = Session(
+            "123", Date(), User(null, null, null), false, notifier, NoopLogger, apiKey
+        )
+    }
+
+    @Test
+    fun generateUnhandledReport() {
+        assertEquals(DeliveryStrategy.STORE_ONLY, event.deliveryStrategy)
+        event.deliveryStrategy = DeliveryStrategy.SEND_IMMEDIATELY
+        assertEquals(DeliveryStrategy.SEND_IMMEDIATELY, event.deliveryStrategy)
+    }
+
+    @Test
+    fun testDeliveryStrategyStoreAndFlush() {
+        // Test ANR case
+        val anrEvent = Event(
+            testException, config,
+            SeverityReason.newInstance(SeverityReason.REASON_ANR),
+            NoopLogger
+        )
+        anrEvent.getErrors().get(0).setErrorClass("ANR")
+        assertEquals(DeliveryStrategy.STORE_AND_FLUSH, anrEvent.getDeliveryStrategy())
+
+        // Test promise rejection case
+        val promiseRejectionEvent = Event(
+            testException, config,
+            SeverityReason.newInstance(SeverityReason.REASON_PROMISE_REJECTION),
+            NoopLogger
+        )
+        assertEquals(DeliveryStrategy.STORE_AND_FLUSH, promiseRejectionEvent.getDeliveryStrategy())
+
+        // Test ANR with modified error class
+        val modifiedAnrEvent = Event(
+            testException,
+            config,
+            SeverityReason.newInstance(SeverityReason.REASON_PROMISE_REJECTION),
+            NoopLogger
+        )
+        modifiedAnrEvent.getErrors().get(0).setErrorClass("ANR")
+        assertEquals(DeliveryStrategy.STORE_AND_FLUSH, modifiedAnrEvent.getDeliveryStrategy())
+    }
```

Comment:
I think these would be better split up into actually separate test cases, instead of just split by comment